configs/dc2.yaml

```diff
@@ -0,0 +1,51 @@
+facility_name: Rubin-LSST
```

Comment:
Yes, maybe simplest to set instrument in the Butler constructor as a governor dimension.

I'm assuming we'd like it such that the default query can be formed from the config file but the command line user can override things like the collection and possibly instrument.

Reply:
The instrument name is not specified in the config file, but comes from DataId. Some dataset types (e.g. coadds) do not have instrument name in their dimensions, I guess coadds can potentially include images from more than one instrument. I can add instrument to the configuration and use it in cases when instrument name is not known from DataId, if you prefer.

Reply:
And instrument name in dc2 repo is `LSSTCam-imSim`. Do you want me to add an option to override instrument name that comes from registry?

Reply:
We were very particular about choosing the name above and not making it seem like LSSTCam data. 

I think the instrument not being included in the query means that it is falling back to the default from the collection. I'm pretty sure that saying `instrument=X, skymap=Y` (if they are given) in the Butler constructor will allow the right thing to happen since they are governor dimensions. If we are doing healpix coadds then they will have a different skymap to the normal patch/tract coadds but I think you are saying that's all fine because they will be in different collections so it will all be sorted out automatically.

I do now wonder if we need the instrument at all, since as you say, given the collection is explicitly given and likely will only have one instrument in it.

Reply:
I think there may be two separate issues here, one is how to filter by instrument name, right now there is no such filtering at all. When I add "where" option we may need to specify governor dimension, but it could be done in the same "where" expression. 

Second issue is what we write to the output file, as I said currently I take it from DataId but not all dataset types have it. And I don't know if we also want to overwrite instrument in some cases even if it is present in a dataset.

Reply:
Thinking about instrument again, I believe what I have now should work fine already:
- if one wants to filter by instrument name then "where" option can be used for that
- overriding instrument name on output is already possible by using `extra_columns` option in the config file, if that option specifies an existing column then its value will be replaced with the value from config file (`extra_columns` is both global and per-dataset).